### PR TITLE
[mariadb][common] Enable VPA for mariadb container

### PR DESCRIPTION
**Pull request by rajivmucheli**
Service owner can set enabled: True to enable, by default its false

### CODE REVIEW

**Review by Nuckal777 — APPROVED** [review]
lgtm

**Comment by rajivmucheli** [comment]
Hi @viktor-kovacs, 

Shall we rollback the minor version mariadb upgrade to allow this commit to be merged ?

**Comment by rajivmucheli** [comment]
FYI @joker-at-work @sebageek, this PR introduces VPA for nova, neutron mariadb deployments.

File: common/mariadb/templates/deployment.yaml [hunk 1]
```diff
@@ -26,7 +26,9 @@ spec:
         prometheus.io/scrape: "true"
         prometheus.io/targets: {{ required ".Values.alerts.prometheus missing" .Values.alerts.prometheus | quote }}
 {{- end }}
-
+{{- if .Values.vpa.enabled }}
+        vpa-butler.cloud.sap/main-container: mariadb
+{{- end }}
```
Comment (joker-at-work) [inline common/mariadb/templates/deployment.yaml:31]: Why do have to enable it explicitly? Does it hurt if it's active by default?

I don't think `vpa.enabled` catches the meaning very good. I'd prefer something like `vpa.set_main_container`.

Comment (rajivmucheli) [reply common/mariadb/templates/deployment.yaml:31]: only nova, neutron, keystone were part of the list. Other services were not.

**Comment by asseco-int-mgmt** [comment]
> Hi @viktor-kovacs,
> 
> Shall we rollback the minor version mariadb upgrade to allow this commit to be merged ?

Hi Rajiv!

Yes, We are going to rolling back today.

Regards,Viktor

File: common/mariadb/templates/deployment.yaml (hunk 1 above)
Comment (joker-at-work) [reply common/mariadb/templates/deployment.yaml]: Doesn't answer the second question, but with the rename of the value, I'm fine.

Comment (Nuckal777) [reply common/mariadb/templates/deployment.yaml]: Should work, maybe make a toggle to remove the annotation instead?

Comment (viktor-kovacs) [reply common/mariadb/templates/deployment.yaml]: An Enable/Disable (toggle) option is more useful even if the default value is enabled or disable (true/false) than and option to remove if you don't want to enable it. In that case nobody won't forget to specify if they want to enable/disable it. Also it helps during templating.

**Review by businessbean — APPROVED** [review]
lgtm with the recommended changes and the added documentation snippet

**Pull request merged**